yktrace/src/hwt/mapper.rs

```diff
@@ -2,39 +2,43 @@
 
 use crate::IRBlock;
 use byteorder::{NativeEndian, ReadBytesExt};
-use fxhash::FxHashMap;
 use hwtracer::{HWTracerError, Trace};
+use intervaltree::{self, IntervalTree};
 use memmap2;
 use object::{Object, ObjectSection};
-use phdrs::objects;
+use once_cell::sync::Lazy;
 use std::{
     convert::TryFrom,
     env, fs,
     io::{prelude::*, Cursor, SeekFrom},
 };
+use ykllvmwrap::symbolizer::Symbolizer;
+use ykutil::addr::code_vaddr_to_off;
 
 const BLOCK_MAP_SEC: &str = ".llvm_bb_addr_map";
+static BLOCK_MAP: Lazy<BlockMap> = Lazy::new(|| BlockMap::new());
 
 /// The information for one basic block, as per:
 /// https://llvm.org/docs/Extensions.html#sht-llvm-bb-addr-map-section-basic-block-address-map
+#[derive(Debug)]
 #[allow(dead_code)]
 struct BlockMapEntry {
     /// Function offset.
     f_off: u64,
     /// Basic block number.
-    bb: u8,
+    bb: usize,
 }
 
 /// Maps (unrelocated) block offsets to their corresponding block map entry.
 pub struct BlockMap {
-    map: FxHashMap<usize, BlockMapEntry>,
+    tree: IntervalTree<u64, BlockMapEntry>,
```

Comment:
I notice that the block index here is a `u64` but above we have `bb: usize`. Are they / should they be the same integer types?

Reply:
The interval tree is mapping ranges of file offsets to block map entries, so the `u64` there is actually a file offset.

Reply:
OK.

Reply:
In a later PR we could wrap `u64` into a `Off` struct, and `uszie` into a `Vaddr` struct for clarity/added type safety?

Reply:
Up to you.